src/optiga/pal/pal_os_event.c

```diff
@@ -56,6 +59,9 @@ void pal_os_event_start(
 void pal_os_event_stop(pal_os_event_t* p_pal_os_event)
 {
     p_pal_os_event->is_event_triggered = 0;
+    if (!scheduler_was_triggered) {
+        timer_remove_task(&TIMER_0, &scheduler);
+    }

```

Comment:
The optiga lib's libusb PAL example removes the timer in `pal_os_event_destroy()` like this:

https://github.com/BitBoxSwiss/optiga-trust-m/blob/5b924c8615f563334136d5a84431f1682cfce3ea/extras/pal/libusb/pal_os_event.c#L287-L289

I am not sure what is going on under the hood yet, but maybe that would be the proper solution?

Reply:
Hmm, yeah I guess what we missed was actually to remove the task in the `destroy` function. That also solves the issue. made the PR much smaller.